packages/web-components/@groupby/elements-products/src/products-base.ts

```diff
@@ -40,6 +52,26 @@ export default class ProductsBase extends Base {
     }
   }
 
+  /**
+   * Requests initial data for the component.
+   *
+   * @param name The name of the initial cached data to return.
+   * @param group The name of the event group that the component belongs to.
+   * @param returnEvent The name of the event under which to dispatch the cached data.
+   */
+  requestInitialData(name: string, group: string, returnEvent: string): void {
+    const payload: CacheRequestPayload = { name, group, returnEvent };
+    this.dispatchElementsEvent<CacheRequestPayload>(CACHE_REQUEST, payload);
+  }
+
+  /**
+   * Generates a string intended to be used as the name of the return event in
+   * cache requests for the component.
+   */
+  get cacheResponseEventName(): string {
+    return `${CACHE_RESPONSE_PREFIX}${this.tagName}-${this.componentId}`;
+  }
+
```

Comment:
@geoffcoutts even though this would mean duplication of code? Until we have another solution for where to house the cache methods isn't a shared class the best place? @jmbtutor @DanielMoniz-GBI @ChantalDesrochers  My ticket indicated for me to try to leverage the base class but I will move it if you feel strongly

Reply:
@geoffcoutts also do you mean both the cacheResponseEventName and the requestInitialData methods (those are both in base but used in products search and sayt)? For clarity is your suggestion to move those both out into the separate files?

Reply:
@geoffcoutts @ssbelle The `ProductsBase` class is not currently related to events, but I don't think its purpose was to avoid events. The reason we have multiple Products classes (that inherit from Base) was to solve a specific problem: how can we allow for product grids that:
a) respond to different events,
b) are maintainable by the developers (us), and
c) are clear for and easy to use by clients?

As far as I can see, if we can avoid duplication between the product grids, we should do so by adding code to `ProductsBase`. If the logic required to do so is complex, we should duplicate it across the separate components (as we did with the events). In this case, the logic is generalizable and seems appropriate for the base class.

Reply:
On further reflection, I agree that utilities for the events can be in `ProductsBase` (and really, this particular getter should probably live in `Base`) as long as `ProductsBase` remains flexible and doesn't require its descendants to send or handle specific events. Having this getter here is compatible with that.

Reply:
@ssbelle @DanielMoniz-GBI @jmbtutor All good. I didn't have any major problems with it going in `ProductsBase`, it just presently didn't conform to what the class was doing. Given that these methods are going to be staying put in `ProductsBase`, does it make sense to also move the response handler there as well?

Reply:
leave

Reply:
done